model/index.js

```diff
@@ -1,15 +1,61 @@
-// const fs = require('fs/promises')
-// const contacts = require('./contacts.json')
+const fs = require('fs/promises')
+const path = require('path')
+const crypto = require('crypto')
 
-const listContacts = async () => {}
+const contactsPath = path.join(__dirname, './contacts.json')
 
-const getContactById = async (contactId) => {}
+const readData = async () => {
+  const result = await fs.readFile(contactsPath, 'utf8')
+  return JSON.parse(result)
+}
+
+const writeData = async (data) => {
+  const stringifiedData = JSON.stringify(data)
+  await fs.writeFile(contactsPath, stringifiedData)
+}
 
-const removeContact = async (contactId) => {}
+const listContacts = async () => {
+  return await readData()
+}
+
+const getContactById = async (contactId) => {
+  const contacts = await readData()
+  const [filteredContact] = contacts.filter(
+    (item) => String(item.id) === contactId
+  )
+  return filteredContact
+}
 
-const addContact = async (body) => {}
+const removeContact = async (contactId) => {
+  const contacts = await readData()
+  const [contactToDelete] = contacts.filter(
+    (item) => String(item.id) === contactId
+  )
+  const contactToDeleteIdx = contacts.indexOf(contactToDelete)
```

Comment:
Вы один раз прошлись по массиву, нашли элемент. А потом второй раз прошлись по массиву, нашли его индекс. Может быть можно найти индекс и по индексу достать этот элемент?

Reply:
Переделал - теперь 1 проход
```suggestion
  const contacts = await readData()
  const contactToDeleteIdx = contacts.findIndex((item) => item.id === contactId)
  const removed = contacts.splice(contactToDeleteIdx, 1)
  await writeData(contacts)

  return removed
}
```